src/app-layer-detect-proto.c

```diff
@@ -686,7 +686,7 @@ static uint32_t AppLayerProtoDetectProbingParserGetMask(AppProto alproto)
         FatalError("Unknown protocol detected - %u", alproto);
     }
 
-    SCReturnUInt(1UL << (uint32_t)alproto);
+    SCReturnUInt((uint32_t)1 << (uint32_t)alproto);
```

Comment:
can we use `BIT_U32(alproto)`?

Reply:
Nope, gives a warning
```
  CC       app-layer-detect-proto.o
app-layer-detect-proto.c:689:18: warning: implicit conversion loses integer precision: 'unsigned long' to 'uint32_t' (aka 'unsigned int') [-Wshorten-64-to-32]
  689 |     SCReturnUInt(BIT_U32(alproto));
```

Reply:
the macro would need the same fix

Reply:
indeed BIT_U16 macro seems to have a fix already